frontend/src/app/connection/DefaultStreamlitEndpoints.test.ts

```diff
@@ -157,16 +178,53 @@ describe("DefaultStreamlitEndpoints", () => {
       axiosMock.restore()
     })
 
-    it("calls the appropriate endpoint", async () => {
+    it("properly constructs the correct endpoint when given a relative URL", async () => {
       axiosMock
-        .onPost("http://streamlit.mock:80/mock/base/path/_stcore/upload_file")
+        .onPost(
+          "http://streamlit.mock:80/mock/base/path/_stcore/upload_file/file_1"
+        )
+        .reply(() => [200, 1])
+
+      const mockOnUploadProgress = (_: any): void => {}
+      const mockCancelToken = axios.CancelToken.source().token
+
+      await expect(
+        endpoints.uploadFileUploaderFile(
+          "/_stcore/upload_file/file_1",
+          MOCK_FILE,
+          "mockWidgetId",
+          "mockSessionId",
+          mockOnUploadProgress,
+          mockCancelToken
+        )
+      ).resolves.toBeDefined()
+
+      const expectedData = new FormData()
+      expectedData.append("sessionId", "mockSessionId")
+      expectedData.append("widgetId", "mockWidgetId")
```

Comment:
in production implementation we should remove `widgetId`, since it is obsolete

Reply:
Removed!